lib/matplotlib/axes/_secondary_axes.py

```diff
@@ -0,0 +1,463 @@
+            self.apply_aspect(pos)
+        else:
+            self.apply_aspect()
+
+        if self._orientation == 'x':
+            bb_axis = self.xaxis.get_tightbbox(renderer)
+        else:
+            bb_axis = self.yaxis.get_tightbbox(renderer)
+        if bb_axis:
+            bb.append(bb_axis)
+
+        bb.append(self.get_window_extent(renderer))
+        _bbox = mtransforms.Bbox.union(
+            [b for b in bb if b.width != 0 or b.height != 0])
+
+        return _bbox
+
+    def set_aspect(self, *args, **kwargs):
+        """
+        Secondary axes cannot set the aspect ratio, so calling this just
+        sets a warning.
+        """
+        cbook._warn_external("Secondary axes can't set the aspect ratio")
+
+    def set_xlabel(self, xlabel, fontdict=None, labelpad=None, **kwargs):
+        """
+        Set the label for the x-axis.
+
+        Parameters
+        ----------
+        xlabel : str
+            The label text.
+
+        labelpad : scalar, optional, default: None
+            Spacing in points between the label and the x-axis.
+
+        Other Parameters
+        ----------------
+        **kwargs : `.Text` properties
+            `.Text` properties control the appearance of the label.
+
+        See also
+        --------
+        text : for information on how override and the optional args work
+        """
+        if labelpad is not None:
+            self.xaxis.labelpad = labelpad
+        return self.xaxis.set_label_text(xlabel, fontdict, **kwargs)
+
+    def set_ylabel(self, ylabel, fontdict=None, labelpad=None, **kwargs):
+        """
+        Set the label for the x-axis.
+
+        Parameters
+        ----------
+        ylabel : str
+            The label text.
+
+        labelpad : scalar, optional, default: None
+            Spacing in points between the label and the x-axis.
+
+        Other Parameters
+        ----------------
+        **kwargs : `.Text` properties
+            `.Text` properties control the appearance of the label.
+
+        See also
+        --------
+        text : for information on how override and the optional args work
+        """
+        if labelpad is not None:
+            self.yaxis.labelpad = labelpad
+        return self.yaxis.set_label_text(ylabel, fontdict, **kwargs)
+
+    def set_color(self, color):
+        """
+        Change the color of the secondary axes and all decorators
+        Parameters
+        ----------
+        color : Matplotlib color
+        """
+
+        if self._orientation == 'x':
+            self.tick_params(axis='x', colors=color)
+            self.spines['bottom'].set_color(color)
+            self.spines['top'].set_color(color)
+            self.xaxis.label.set_color(color)
+        else:
+            self.tick_params(axis='y', colors=color)
+            self.spines['left'].set_color(color)
+            self.spines['right'].set_color(color)
+            self.yaxis.label.set_color(color)
+
+
+_secax_docstring = '''
+Warnings
```

Comment:
warning could use a rst directive, as in https://numpydoc.readthedocs.io/en/latest/format.html#other-points-to-keep-in-mind.

Reply:
Huh, the docstrings aren't rendering the interpolation at all...

Reply:
https://18201-1385122-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/api/_as_gen/matplotlib.axes.Axes.secondary_xaxis.html#matplotlib.axes.Axes.secondary_xaxis  

So not sure why this should be an rst directive versus the numpy doc heading.  Is there a difference?

Reply:
Looking at numpydoc's implementation they just internally translate warnings sections to `.. warnings:`, but also put the sections in their own (consistent) order (as seen in the rendered docs); I mildly prefer the latter as you can have them wherever you want or nested under other constructs (e.g. a bullet list) but it doesn't matter too much.

Reply:
ummm, so which do you prefer? I dont' really care - easier to remember the numpydoc way than the rst way

Reply:
Strictly personally I prefer the rst directive, but again anything works here (though you will need to use the directive if you want the warning to show up *above* the parameters in the rendered html).

Reply:
OK, I'll leave as is - I can see some argument for the warning at the top, but don't think its a big deal.